### ISSUE TITLE
Meet with Medallia for upgrade walkthrough (March 6)

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/va.gov-team#77545]

**Comment by eli-oat** [comment]
No updates were shown, but a document was screen shared that detailed the accessibility updates that are *coming* -- not arrived, yet. 

Once the accessibility improvements are available, I offered to have CAIA help validate/review the improvements.

**Comment by coforma-jamie** [comment]
Sharing Medallia's [Digital Form 2.0 - Product Overview](https://github.com/department-of-veterans-affairs/va.gov-team/files/14526804/Digital.Form.2.0.-.Product.Overview.docx) doc for visibility

Without seeing this (or the previous implementation) with my own eyes, I'm making some educated guesses here. But based on their document, here's what I think we need to look out for:

### Proper use of `<label>`
I'm guessing that previously, questions were marked up using `<H2>` and `<H3>` tags. They say that those tags 
> "will be removed and replaced with a compliant markup." 

Note that the headings themselves aren't an issue, it's the lack of a `<label>` that's the problem. It's entirely possible - and accessible! - to make the question BOTH a heading and a `<label>`, such as in this [VADS component](https://design.va.gov/storybook/?path=/docs/components-va-checkbox-group--label-header#label-header).

**What we'll need to look out for:** 
- Do all single-input fields - text input, textarea, number fields - have a properly associated label?
- Are those labels ALSO headings (at the appropriate heading level) where appropriate?
- I'm also going to take a close look at grouped inputs, like checkboxes and radio buttons. Are they in a proper `<fieldset>`, with a `<legend>`?

### Color contrast
Medallia notes two changes to color contrast:
- Border color will meet 3:1 contrast (this aligns with WCAG 2.1 AA for [non-text UI elements](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html))
- "Accept/Cancel" buttons will update the default color "to meet the contrast requirements." They'll also add a default hover color change.

**What we'll need to look out for:**
- What standards are they using for the Accept/Cancel buttons and button hover color? And do they meet those standards?
   - Text has stricter contrast standards than non-text. WCAG 2.1 level AA requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text (18 point text or 14 point bold text). AAA has stricter standards.
- I'd like to test these using the [Colorblindly](https://chromewebstore.google.com/detail/colorblindly/floniaahmccleoclneebhhmnjgdfijgg) Chrome extension as well.

### Required fields
Medallia says that they will remove "unnecessary Aria-Required labels from survey questions."

I assume that they're using a different way to note required fields, probably the [HTML5 required attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required). Removing extraneous aria is always a good idea!

**What we'll need to look out for:**
- What method are they using for required fields? (Is my assumption correct?)
- Is there a visual indicator of required fields? What is that indicator? Is it ONLY color? (It needs to be more than color!)
- How are they displaying errors to users? Are errors accessible? Do they actually NEED to use aria to handle errors correctly?
